jetbrains-gateway/src/software/aws/toolkits/jetbrains/gateway/welcomescreen/WorkspaceDataRetriever.kt

```diff
@@ -147,7 +148,9 @@ class WorkspaceDataRetriever(
     }
 
     fun markWorkspaceAsDirty(identifer: WorkspaceIdentifier) {
-        dirtyWorkspaces.add(identifer)
+        if (CawsSpaceTracker.getInstance().lastSpaceName() == identifer.project.space) {
+            dirtyWorkspaces.add(identifer)
+        }
         updateAlarm.cancelAllRequests()
         pollForUpdate()
```

Comment:
should be complete no-op
```suggestion
        if (CawsSpaceTracker.getInstance().lastSpaceName() != identifer.project.space) {
            return
        }
        dirtyWorkspaces.add(identifer)
        updateAlarm.cancelAllRequests()
        pollForUpdate()
```